client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -125,9 +126,16 @@ export default function getThankYouPageUrl( {
 	jetpackTemporarySiteId,
 	adminPageRedirect,
 	domains,
+	saasProductRedirectUrl,
 }: PostCheckoutUrlArguments ): string {
 	debug( 'starting getThankYouPageUrl' );
 
+	// In the case when SaaS redirect URL is passed as argument, we want to redirect
+	// user to vendor application after successful purchase.
+	if ( saasProductRedirectUrl ) {
+		return saasProductRedirectUrl;
```

Comment:
Just out of curiosity, is there no way to do this with the existing `redirectTo` parameter?

When this function is called, could we do something like 

```js
getThankYouPageUrl( { 
  //...
  redirectTo: saasProductRedirectUrl ?? redirectTo,
  //...
} )
```

Or was the purpose of this parameter to sidestep all the guards below?